src/webots/nodes/WbJoint.cpp

```diff
@@ -296,8 +296,8 @@ void WbJoint::updateJointAxisRepresentation() {
 }
 
 const QString WbJoint::urdfName() const {
-  if (positionSensor() && positionSensor()->findSFString("name"))
-    return getUrdfPrefix() + positionSensor()->findSFString("name")->value();
+  if (motor() && motor()->findSFString("name"))
+    return getUrdfPrefix() + motor()->motor("name")->value();
```

Comment:
```suggestion
  if (motor() && motor()->findSFString("name"))
    return getUrdfPrefix() + motor()->motor("name")->value();
  else if (positionSensor() && positionSensor()->findSFString("name"))
    return getUrdfPrefix() + positionSensor()->findSFString("name")->value();
```

Just to handle the case of a non-actuated joint.